### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by Paskab012**
Hello 👋 , 
In this PR, I have Implement the following
## Interface
- Created a class Nameable.
- Implemented a method called correct_name that will raise a NotImplementedError.
- Turned the Person class to Nameable
- Made sure that the Person class inherits from Nameable
- Made sure that this class has a method correct_name implemented. It should simply return the name attribute.

   ### Prepare base Decorator

- Made sure that it inherits from Nameable.
In the constructor assign a nameable object from params to an instance variable.
- Implemented the correct_name method that returns the result of the correct_name method of the @nameable.
- Prepared CapitalizeDecorator and TrimmerDecorator
   - For the CapitalizeDecorator:
      - Created a class that inherits from the base Decorator class.
      - Implemented a method correct_name that capitalizes the output of @nameable.correct_name.
   - For the TrimmerDecorator:
      - Created a class that inherits from the base Decorator class.
      - Implemented a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: trimmer_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  attr_accessor :nameable
+
+  def correct_name
+    @nameable.correct_name
+  end
```
Comment (Kingstalux) [inline trimmer_decorator.rb:13]: - [ ] Great job so far 👍 Kindly implement the `TrimmerDecorator` class which should check the length of a name and trim the length if it's greater than 10. You've implemented the `decorator` class twice instead. Kindly rectify this issue 👍

**Review by Kingstalux — APPROVED** [review]

File: trimmer_decorator.rb (hunk 1 above)
Comment (Kingstalux) [inline trimmer_decorator.rb]: - [ ] Great job so far 👍 Kindly implement the `TrimmerDecorator` class which should check the length of a name and trim it if it is greater than 10. You've instead implemented the decorator class twice. Kindly rectify this issue 👍

**Review by Kingstalux — CHANGES_REQUESTED** [review]
Hi @Paskab012, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

149637855-7722d04c-b45d-4185-925f-7de047106461

## Highlights 👏 
✅ Good pull request description 🔥 
✅ Professional README file 💯 
✅ Linters are passing 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rustam-tolipov — APPROVED** [review]
## Hi @Paskab012 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 🎉 🎉 🎉 🎉 

Cheers and Happy coding!👏👏👏

Developer-512px-10

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**